### PR TITLE
Update ember-and-json-api.md

### PR DESCRIPTION
**Pull request by Ikariusrb**
#### Purpose
Update the ember integration documentation

#### Changes
Update documentation for how to get ember to use include directives
Add documentation on how to update controllers to tell Rails we are consuming and generating jsonapi data.

#### Additional helpful information

Removed ember-data adapter change to support include directives, as it's now built-in.  Updated the documentation for how to add include directives to ember store queries, and added documentation covering how to tell Rails that we are consuming and generating jsonapi data

### CODE REVIEW

File: docs/integrations/ember-and-json-api.md [hunk 1]
```diff
@@ -38,12 +38,40 @@ You will also want to set the `key_transform` to `:unaltered` since you will adj
...
 ```
 
-Lastly, in order to properly handle JSON API responses, we need to register a JSON API renderer, like so:
+In order to properly handle JSON API responses, we need to register a JSON API renderer, like so:
 
 ```ruby
 # config/initializers/active_model_serializers.rb
 require 'active_model_serializers/register_jsonapi_renderer'
 ```
+Rails also requires your controller to tell it that you accept and generate JSONAPI data.  To do that, you use `respond_to` in your controller handlers.  You can add `ActionController::MimeResponds` to your application controller to enable this:
```
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: why is this? was this mentioned in the slack convo you had with @bf4? might be a good idea to state *why* you *need* respond_to.

File: docs/integrations/ember-and-json-api.md [hunk 2]
```diff
@@ -38,12 +38,40 @@ You will also want to set the `key_transform` to `:unaltered` since you will adj
...
+        format.jsonapi { render jsonapi: @post, status: :created, location: @post }
+      else
+        format.jsonapi { render jsonapi: @post.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+  
+    # Only allow a trusted parameter "white list" through.
+    def post_params
+      ActiveModelSerializers::Deserialization.jsonapi_parse!(params, only: [:title, :body] )
```
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: is the exclamation point required? I'm using this without the exclamation point. Exclamation point implies mutation.

File: docs/integrations/ember-and-json-api.md [hunk 3]
```diff
@@ -67,18 +95,6 @@ export default  DS.JSONAPIAdapter.extend({
     return pluralize(underscored);
   },
 
-  // allows queries to be sent along with a findRecord
```
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: no longer needed since Ember 2.5!!!! (I think it was 2.5)

File: docs/integrations/ember-and-json-api.md [hunk 4]
```diff
@@ -102,18 +118,16 @@ export default DS.JSONAPISerializer.extend({
 
 ```
 
+
 ## Including Nested Resources
 
-Previously, `store.find` and `store.findRecord` did not allow specification of any query params.
-The ActiveModelSerializers default for the `include` parameter is to be `nil` meaning that if any associations are defined in your serializer, only the `id` and `type` will be in the `relationships` structure of the JSON API response.
+Ember-data can request related records by using `include`.  To tell it to make that request, use the form below when requesting records.
```
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: Ember Data is referred to as a "Ember Data" rather than "Ember-data", see: https://github.com/emberjs/data

File: docs/integrations/ember-and-json-api.md [hunk 5]
```diff
@@ -102,18 +118,16 @@ export default DS.JSONAPISerializer.extend({
...
 For more on `include` usage, see: [The JSON API include examples](./../general/adapters.md#JSON-API)
 
-With the above modifications, you can execute code as below in order to include nested resources while doing a find query.
-
 ```javascript
-store.findRecord('post', postId, { adapterOptions: { query: { include: 'comments' } } });
+store.findRecord('post', postId, { include: 'comments' } );
 ```
-will generate the path `/posts/{postId}?include='comments'`
+which will generate the path /posts/{postId}?include='comments'
```
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: line 134 should be changed to `render jsonapi: ...`

File: docs/integrations/ember-and-json-api.md (hunk 4 above)
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: I would reword "use the form below" it sounds awkward, imo.

File: docs/integrations/ember-and-json-api.md (hunk 2 above)
Comment (bf4) [inline docs/integrations/ember-and-json-api.md]: I prefer it rather than silent failure and invalid data

File: docs/integrations/ember-and-json-api.md (hunk 1 above)
Comment (bf4) [inline docs/integrations/ember-and-json-api.md]: This is more like the following story:

`render jsonapi: foo`

Client: Hello. I am sending you JSONAPI data.  I'll accept yours answers only as JSON API data
Server: Here's some JSON API data

`respond_to do |format| format.jsonapi { render jsonapi: foo } end`

Client: Hello. I am sending you JSONAPI data.  I'll accept yours answers only as JSON API data
Server: You want JSON API data? Here's some JSON API data

File: docs/integrations/ember-and-json-api.md (hunk 2 above)
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: ah, that makes sense. 
In my app, I disabled document validating, cause I'm using embedded documents. I think that was pre-`!` though. I don't remember much of that code and what is called when.

File: docs/integrations/ember-and-json-api.md (hunk 1 above)
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: I like this story.

Comment (bf4) [inline docs/integrations/ember-and-json-api.md]: Which means

> Rails also requires your controller to tell it that you accept and generate JSONAPI data

Isn't quite true. Rails even has an [ignore_accept_header](https://github.com/rails/rails/search?utf8=%E2%9C%93&q=ignore_accept_header&type=Code)

**Comment by bf4** [comment]
You may want to rebase this off of master.

File: docs/integrations/ember-and-json-api.md (hunk 3 above)
Comment (bf4) [inline docs/integrations/ember-and-json-api.md]: should this be included as an comment somewhere? People are still on 1.13

File: docs/integrations/ember-and-json-api.md [hunk 6]
```diff
@@ -102,18 +118,16 @@ export default DS.JSONAPISerializer.extend({
 
 ```
 
+
 ## Including Nested Resources
```
Comment (bf4) [inline docs/integrations/ember-and-json-api.md]: Would be great to mention the whole 'embedded resources' controversy here and how it's not JSON API so we don't do it, but ....

File: docs/integrations/ember-and-json-api.md (hunk 3 above)
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: I'm not sure if 1.13 is officially supported other than some bug fixes.

File: docs/integrations/ember-and-json-api.md (hunk 6 above)
Comment (NullVoxPopuli) [inline docs/integrations/ember-and-json-api.md]: yeah, maybe mention the difference between including, and embedding, and that there are ways to embed, such as: https://github.com/NullVoxPopuli/aeonvera/blob/master/config/initializers/active_model_serializers.rb#L18

**Comment by NullVoxPopuli** [comment]
@Ikariusrb do you think you can finish this up?
